src/main/java/com/hillel/items_exchange/mapper/CategoryMapper.java

```diff
@@ -0,0 +1,82 @@
+package com.hillel.items_exchange.mapper;
+
+import com.hillel.items_exchange.dao.SubcategoryRepository;
+import com.hillel.items_exchange.dto.CategoryControllerDto;
+import com.hillel.items_exchange.dto.SubcategoryControllerDto;
+import com.hillel.items_exchange.model.Category;
+import com.hillel.items_exchange.model.Subcategory;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class CategoryMapper {
```

Comment:
Do we really need this class? I think that ModelMapper might help us with conversation DTO to model and vice versa.